driver-core/src/test/functional/com/mongodb/internal/connection/TestCommandListener.java

```diff
@@ -37,25 +37,45 @@
 
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.locks.Condition;
 import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReentrantLock;
 
 import static com.mongodb.ClusterFixture.TIMEOUT;
+import static java.util.Arrays.asList;
 import static java.util.Collections.emptyList;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertTrue;
 
 public class TestCommandListener implements CommandListener {
+    private static final Set<String> SENSITIVE_COMMANDS = new HashSet<>(
+            asList(
+                    "authenticate",
+                    "saslStart",
+                    "saslContinue",
+                    "getnonce",
+                    "createUser",
+                    "updateUser",
+                    "copydbgetnonce",
+                    "copydbsaslstart",
+                    "copydb"));
+    private static final Set<String> SENSITIVE_HANDSHAKE_COMMANDS = new HashSet<>(
+            asList(
+                    "ismaster",
+                    "isMaster",
+                    "hello"));
```

Comment:
> uncomfortable with using production code as part of an assertion, since that's the code that is under test

In our case the sets `InternalStreamConnection.SECURITY_SENSITIVE_COMMANDS` and `InternalStreamConnection.SECURITY_SENSITIVE_HELLO_COMMANDS` are not being tested by `TestCommandListener`. If we had an external source of truth from which the test could automatically extract these sets and then compare the extracted data with the data hardcoded in our sets, that would be a test for the hardcoded data, and such a test would warn us that the data has changed in the source of truth, and we must update our hardcoded data accordingly.

`TestCommandListener` as is simply duplicates the sets `InternalStreamConnection.SECURITY_SENSITIVE_COMMANDS`/`InternalStreamConnection.SECURITY_SENSITIVE_HELLO_COMMANDS` and introduces a manual step we have to do each time we need to modify these sets. This step adds work without providing anything useful, because all we know is that the sets in `InternalStreamConnection` and `TestCommandListener` must be equal, but neither is the source of truth and both are managed by the same people (the Java driver maintainers).

Reply:
Done